src/shared/lib/utils/dateTime.ts

```diff
@@ -79,3 +79,18 @@ export function getLast7Dates() {
     .map(i => subDays(now, i))
     .reverse();
 }
+
+export const buildDateFromDto = (
+  dto: string | null | undefined,
+): Date | null => {
+  if (dto === null || dto === undefined) {
+    return null;
+  }
+  const result = new Date(`${dto}T00:00`);
+
+  if (result.getFullYear() === 1 || result.getFullYear() === 9999) {
```

Comment:
Please put the condition into a separate variable.

Reply:
this is temp, waiting for fix BE